### PR TITLE
feat(sql): New syntax 'drop table if exists table-name' to not fail when the table does not exist

### PR DESCRIPTION
**Pull request by marregui**
Before this change, the statement:

```
drop table mytable
```
would fail, which meant having to do a try/catch in client code.

After this change, we support in addition the statement:

```
drop table if exists mytable
```
which will drop the table if exists, and do nothing when not.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|23|23|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|23|23|100.00%|

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java [hunk 1]
```diff
@@ -2673,7 +2674,7 @@ public void testPreparedStatementWithBindVariablesSetWrongOnDifferentConnection(
                 }
 
                 try (final Connection connection = getConnection(false, false);
-                     PreparedStatement statement = connection.prepareStatement("drop table xts")) {
+                     PreparedStatement statement = connection.prepareStatement("drop table if exists xts")) {
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java:2677]: We must not change existing tests to conform with new implementation. Existing tests to ensure non-regressive behaviour.

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|23|23|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|23|23|100.00%|

File: core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java (hunk 1 above)
Comment (marregui) [reply core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java]: I have added a test in DropTableTest as well

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 19 / 19 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|19|19|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 19 / 19 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|19|19|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 19 / 19 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCompiler.java|19|19|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**